lingua_franca/lang/parse_de.py

```diff
@@ -900,107 +1032,91 @@ def date_found():
+
+    input_str = input_str.lower()
+    numerator = 1
+    prev_number = 0
+    denominator = False
+
+    # first check if is a fraction containing a char (eg "2/3")
+    _bucket = input_str.split('/')
+    if look_for_fractions(_bucket):
+        numerator = float(_bucket[0])
+        denominator = float(_bucket[1])
+
+    if not denominator:
+        for fraction in _STRING_FRACTION.keys():
+            if fraction in input_str and not denominator:
+                denominator = _STRING_FRACTION.get(fraction)
+                _bucket = input_str.split(fraction)
+
+        if len(_bucket) > 1:
+            if not _STRING_NUM.get(_bucket[0], False):
+                #acount for eineindrittel
+                for numstring, number in _STRING_NUM.items():
+                    if _bucket[0].endswith(numstring):
+                        prev_number = _STRING_NUM.get(
+                            _bucket[0].replace(numstring, "", 1), False)
+                        numerator = number
+            else:
+                numerator = _STRING_NUM.get(_bucket[0])
 
-    return False
+    if denominator:
+        return prev_number + (numerator / denominator)
+    else:
+        return False
 
 
 def is_ordinal_de(input_str):
     """
     This function takes the given text and checks if it is an ordinal number.
-
     Args:
         input_str (str): the string to check if ordinal
     Returns:
         (bool) or (float): False if not an ordinal, otherwise the number
         corresponding to the ordinal
-
     ordinals for 1, 3, 7 and 8 are irregular
-
-    only works for ordinals corresponding to the numbers in _DE_NUMBERS
-
+    only works for ordinals corresponding to the numbers in _STRING_NUM
     """
+    val = _STRING_LONG_ORDINAL.get(input_str.lower(), False)
+    # account for numbered ordinals
+    if not val and input_str.endswith('.') and is_numeric_de(input_str[:-1]):
+        val = input_str
+    return val
 
-    lowerstr = input_str.lower()
-
-    if lowerstr.startswith("erste"):
-        return 1
-    if lowerstr.startswith("dritte"):
-        return 3
-    if lowerstr.startswith("siebte"):
-        return 7
-    if lowerstr.startswith("achte"):
-        return 8
-
-    if lowerstr[-3:] == "ste":  # from 20 suffix is -ste*
-        lowerstr = lowerstr[:-3]
-        if lowerstr in _DE_NUMBERS:
-            return _DE_NUMBERS[lowerstr]
-
-    if lowerstr[-4:] in ["ster", "stes", "sten", "stem"]:
-        lowerstr = lowerstr[:-4]
-        if lowerstr in _DE_NUMBERS:
-            return _DE_NUMBERS[lowerstr]
-
-    if lowerstr[-2:] == "te":  # below 20 suffix is -te*
-        lowerstr = lowerstr[:-2]
-        if lowerstr in _DE_NUMBERS:
-            return _DE_NUMBERS[lowerstr]
 
-    if lowerstr[-3:] in ["ter", "tes", "ten", "tem"]:
-        lowerstr = lowerstr[:-3]
-        if lowerstr in _DE_NUMBERS:
-            return _DE_NUMBERS[lowerstr]
+def get_ordinal_index(input_str : str, type_: type = str):
```

Comment:
can these functions that are not shared across languages start with an underscore? `def _get_ordinal_index`

Reply:
or if you think some method is useful it can be added as a new function to be implemented per language